### ISSUE TITLE
HTTP fetch of fs-migrations should use CAR

### ISSUE THREAD

**Issue by lidel** [ipfs/kubo#9159]
Version: 0.14.x

Fetching migration data from IPFS was added in https://github.com/ipfs/kubo/pull/8064, but we did not have https://github.com/ipfs/go-ipfs/pull/8758 at the time and HTTP fetch is still delegating trust to the gateway.

I consider that a bug: migrations should be fetched in trustless mode as a CAR by requesting them as `?format=car` and verifying every block before applying the migration. 

This allows us to use third-party gateways af fallback and/or in scenarios where `ipfs.io` is blocked by ISP etc.

**Comment by BigLep** [comment]
@Jorropo will tackle this after some related light-client work he is doing.

**Comment by BigLep** [comment]
@lidel : is this a duplicate of https://github.com/ipfs/kubo/issues/8851 ?

I see there is also this PR: https://github.com/ipfs/kubo/pull/9250

**Comment by lidel** [comment]
I've closed https://github.com/ipfs/kubo/issues/8851 as a duplicate (this one has more context).

**Comment by lidel** [comment]
[IPIP-402](https://specs.ipfs.tech/ipips/ipip-0402/) shipped. Once ipfs.io upgrades to Kubo 0.21 or later, we will be able to fetch `/ipfs/cid/path/file` and verify it.

I don't think we should block on IPNS record resolution, as we do not use it for migrations, we have hardcoded CID, so we only care about `/ipfs/` paths.
